RecoTracker/LST/plugins/LSTOutputConverter.cc

```diff
@@ -65,8 +73,14 @@ LSTOutputConverter::LSTOutputConverter(edm::ParameterSet const& iConfig)
       seedCreator_(SeedCreatorFactory::get()->create("SeedFromConsecutiveHitsCreator",
                                                      iConfig.getParameter<edm::ParameterSet>("SeedCreatorPSet"),
                                                      consumesCollector())),
-      trajectorySeedPutToken_(produces<TrajectorySeedCollection>()),
-      trackCandidatePutToken_(produces<TrackCandidateCollection>()),
+      trajectorySeedPutToken_(produces<TrajectorySeedCollection>("")),
+      trajectorySeedpLSPutToken_(produces<TrajectorySeedCollection>("pLSTSsLST")),
+      trackCandidatePutToken_(produces<TrackCandidateCollection>("")),
+      trackCandidatepTCPutToken_(produces<TrackCandidateCollection>("pTCsLST")),
+      trackCandidateT5TCPutToken_(produces<TrackCandidateCollection>("t5TCsLST")),
+      trackCandidateNopLSTCPutToken_(produces<TrackCandidateCollection>("nopLSTCsLST")),
+      trackCandidatepTTCPutToken_(produces<TrackCandidateCollection>("pTTCsLST")),
+      trackCandidatepLSTCPutToken_(produces<TrackCandidateCollection>("pLSTCsLST")),
```

Comment:
aren't (at least a few of) these mutually exclusive by use case?
As is, doesn't this about triplicate the output of LST?
Memory is usually not unlimited; so, producing objects just in case is not a good strategy.

For now, perhaps some kind of `// FIXME: need to make creation configurable` would be good enough

Reply:
That is right. The minimal set to have would be `t5TCsLST`, `pTTCsLST` and `pLSTCsLST`. Removing the rest will need a bit of rearranging but I can definitely do it now, to set an example for the future.